tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -501,9 +501,44 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
                       CublasLtMatmulMaybeF8(&existing_gemm).WithOneUser())
                       .WithOneUser(),
                   m::Broadcast(&bias, m::Op())))) {
-      TF_ASSIGN_OR_RETURN(
-          bool was_fused,
-          FuseVectorBiasAdd(instr, bias, existing_gemm, optional_slice));
+      HloInstruction *optional_bias_f16 = nullptr;
+      if (existing_gemm->custom_call_target() == kCublasLtMatmulF8CallTarget &&
+          bias->shape().element_type() == F32) {
```

Comment:
This logic should be in `FuseVectorBiasAdd` itself, after the other checks in `FuseVectorBiasAdd` that ensure the bias can be fused. This means you don't need to add another argument to `FuseVectorBiasAdd`.

Also there should be a comment for this logic. Such as:

"cuBLAS LT does not support FP32 biases on matmuls with FP8 inputs, even if the matmul output is FP32. We do not unconditionally convert the bias to a supported precision (F16 or BF16) because this lowers precision. Instead, we only fuse the bias if the bias itself is a convert from F16 or BF16, fusing the input of the convert instruction to the matmul."

Reply:
Suggested by @philipphack , it's better to isolate the additional logic from `FuseVectorBiasAdd` like in other cases where we have optional unary ops. Will update the comment.

Reply:
`handleAdd` is easier to read when it consists of solely of matches followed by a small amount of code to do the conversion. So it makes sense to move this to `FuseVectorBiasAdd`, even if some other cases handle unary ops outside the conversion method. 

And some cases handle optional unary ops inside the conversion method, like the Abs op in `F8ConvertD`.